Source/santad/SNTSyncdQueue.mm

```diff
@@ -37,11 +42,109 @@ - (instancetype)init {
   self = [super init];
   if (self) {
     _uploadBackoff = std::make_unique<SantaCache<std::string, NSDate *>>(256);
-    _syncdQueue = dispatch_queue_create("com.northpolesec.syncd_queue", DISPATCH_QUEUE_SERIAL);
+    _syncdQueue = dispatch_queue_create("com.northpolesec.syncd_queue",
+                                        DISPATCH_QUEUE_SERIAL_WITH_AUTORELEASE_POOL);
   }
   return self;
 }
 
+/// On each call to reassessSyncServiceConnection, a timer will be created to
+/// run ~1s in the future. If a pending timer exists when this method is
+/// called, it is first cancelled and a new timer is created. This is done to
+/// coalesce bursty calls (e.g. in response to a new configuration that makes
+/// changes to multiple values that affect the sync service running state).
+/// When the timer event handler fires, it will check the current config
+/// state, as well as relevant config deltas, to determine whether the sync
+/// service should be started, bounced, or stopped.
+- (void)reassessSyncServiceConnection {
+  WEAKIFY(self);
+  dispatch_sync(self.syncdQueue, ^{
+    if (self.timer) {
+      dispatch_source_cancel(self.timer);
+    }
+
+    self.timer = dispatch_source_create(DISPATCH_SOURCE_TYPE_TIMER, 0, 0, self.syncdQueue);
+    dispatch_source_set_timer(self.timer,
+                              dispatch_time(DISPATCH_TIME_NOW, 1 * NSEC_PER_SEC),  // 1 second delay
+                              DISPATCH_TIME_FOREVER,                               // won't repeat
+                              100 * NSEC_PER_MSEC);
+
+    // WEAKIFY(self);
+    dispatch_source_set_event_handler(self.timer, ^{
+      STRONGIFY(self);
+      SNTConfigurator *configurator = [SNTConfigurator configurator];
+
+      // If the SyncBaseURL was added or changed, and a connection already
+      // exists, it must be bounced.
+      if (self.previousSyncBaseURL &&
```

Comment:
I believe we're using `.length` to check if the string is valid elsewhere. in the codebase.

Reply:
It won't return nil even if line 819 didn't exist, `URLWithString:` returns a valid object even for an empty string 🤦

Reply:
I partially take this back - I'm going to ensure that the new SyncBaseURL cannot be empty, and if so, force it to nil (tho after the configurator fix, this shouldn't be possible). Then will keep the existing nil check as is because it makes more sense and is (imho) cleaner.